qubesmanager/global_settings.py

```diff
@@ -251,6 +297,50 @@ def __init_updates__(self):
         self.enable_updates_all.clicked.connect(self.__enable_updates_all)
         self.disable_updates_all.clicked.connect(self.__disable_updates_all)
 
+        repos = dict()
+        for i in _run_qrexec_repo('qubes.repos.List').split('\n'):
+            l = i.split('\0')
+            # Keyed by repo name
+            d = repos[l[0]] = dict()
+            d['prettyname'] = l[1]
+            d['enabled'] = l[2] == 'enabled'
+
+        if repos['qubes-dom0-unstable']['enabled']:
+            self.dom0_updates_repo.setCurrentIndex(3)
+        elif repos['qubes-dom0-current-testing']['enabled']:
+            self.dom0_updates_repo.setCurrentIndex(2)
+        elif repos['qubes-dom0-security-testing']['enabled']:
+            self.dom0_updates_repo.setCurrentIndex(1)
+        elif repos['qubes-dom0-current']['enabled']:
+            self.dom0_updates_repo.setCurrentIndex(0)
+        else:
+            raise Exception('Cannot detect enabled dom0 update repositories')
+
+        if repos['qubes-templates-itl-testing']['enabled']:
+            self.itl_tmpl_updates_repo.setCurrentIndex(1)
+        elif repos['qubes-templates-itl']['enabled']:
+            self.itl_tmpl_updates_repo.setCurrentIndex(0)
+        else:
+            raise Exception('Cannot detect enabled ITL template update '
+                            'repositories')
+
+        if repos['qubes-templates-community-testing']['enabled']:
+            self.comm_tmpl_updates_repo.setCurrentIndex(2)
+        elif repos['qubes-templates-community']['enabled']:
+            self.comm_tmpl_updates_repo.setCurrentIndex(1)
+        else:
+            self.comm_tmpl_updates_repo.setCurrentIndex(0)
+
+        self.dom0_updates_repo.currentIndexChanged.connect(
+            _handle_dom0_updates_combobox
+        )
+        self.itl_tmpl_updates_repo.currentIndexChanged.connect(
+            _handle_itl_tmpl_updates_combobox
+        )
+        self.comm_tmpl_updates_repo.currentIndexChanged.connect(
+            _handle_comm_tmpl_updates_combobox
+        )
```

Comment:
All other settings here and in VM settings are applied on OK/Apply click. This shouldn't be an exception.

Reply:
Nice catch; fixed.